### DISCUSSION TITLE
[psycopg3] Cursor.copy with an id column as serial primary key?

### DISCUSSION THREAD

**Discussion by GollyJer** [psycopg/psycopg#440] [Q&A] [answered]
In v2 you could import from a csv without specifying a table's serial id like... 

```python
cur.copy_from(csv_file, 'my_table', columns=('thing1', 'thing2'))
```

In v3 `cursor.copy` has no `columns` parameter. How do we copy in v3 without the column id? Thanks!

**Comment by dvarrazzo** [comment, marked as answer]
You can use
```python

with cur.copy("COPY my_table (thing1, thing2) FROM STDIN") as copy:
    while (data := csv_file.read(8192)):
        copy.write(data)
```

`cur.coyp()` in psycopg 3 is similar to `copy_expert()` in psycopg2 and allows to use [all the features supported by PostgreSQL COPY](https://www.postgresql.org/docs/current/sql-copy.html).
